pyramid_chameleon/precompile.py

```diff
@@ -10,21 +10,29 @@
 import sys
 import logging
 import optparse
+from multiprocessing import Pool
 
 import chameleon.config
 from pyramid_chameleon.zpt import PyramidPageTemplateFile
 
+def _compile_one(args):
+    fullpath = args[0]
+    template_factory = args[1]
+    return compile_one(fullpath, template_factory)
+
 def compile_one(fullpath, template_factory=PyramidPageTemplateFile):
-    assert chameleon.config.CACHE_DIRECTORY is not None
-    template = template_factory(fullpath, macro=None)
-    template.cook_check()
+    try:
+        assert chameleon.config.CACHE_DIRECTORY is not None
```

Comment:
I'd prefer this to be done once, rather than over and over again in a function that is going to be called many times.

Reply:
This is the same as the original processing. ([link to source code](https://github.com/Pylons/pyramid_chameleon/pull/20/files#diff-05ff59a367b28454dbc7de20f6e538dfR17))

I assume that I use it as a single method of a module, not as a command, and I think that `chameleon.config.CACHE_DIRECTORY` is being checked here.
And, I think `assert chameleon.config.CACHE_DIRECTORY is not None` is low overhead so I do not need to delete it in particular.